conda/recipes/cudf/meta.yaml

```diff
@@ -51,6 +51,7 @@ requirements:
     - nvtx >=0.2.1
     - packaging
     - cachetools
+    - ptxcompiler  # CUDA enhanced compatibility. See https://github.com/rapidsai/ptxcompiler
```

Comment:
We'd have to bump the `numba` version to `>=0.54` too since `ptxcompiler` requires it.

Reply:
I made the change, but would  conda not have recognized this from `ptxcompiler`'s meta.yaml? https://github.com/rapidsai/ptxcompiler/blob/main/conda/recipes/ptxcompiler/meta.yaml

Reply:
It should, this was just for consistency though. Upon looking up `rmm` we seem to be requiring a more outdated `numba` there. Should we just leave it as is?

Reply:
IIRC Numba 0.53 didn't disallow installation with NumPy 1.20 (because it wasn't released / we didn't know it had an issue with Numba) then Numba 0.54 added a constraint to prevent it being installed with 1.20 - a user could set up a cudf environment with Numba 0.53.1 and NumPy 1.20, then fail to be able to install ptxcompiler later (e.g. if they want to use a new toolkit with their existing driver) because it won't then be able to install Numba 0.54 with the already-installed NumPy 1.20. So bumping our requirement to Numba 0.54 avoids that particular problematic scenario (though it's probably not one in which a great deal of users end up).

Reply:
> It should, this was just for consistency though. Upon looking up rmm we seem to be requiring a more outdated numba there. Should we just leave it as is?

Hmm, it's going to be hard to sync between cuDF and RMM dependencies in this way. Sounds like a bigger problem we need to think about outside of this PR?